packages/yoroi-extension/chrome/extension/ergo-connector/types.js

```diff
@@ -83,16 +85,16 @@ export function asBox(
 
       {
         // we just want to validate that the boxID matches the content
-        const wasmBox = wasmInstance.ErgoBox.from_json(JSON.stringify(boxJson));
+        const wasmBox = wasmInstance.ErgoBox.from_json(JSONBigInt.stringify(boxJson));
```

Comment:
Does the sigma-rust JSON parser interpret this correctly? From what I've read JSON doesn't specify the range but a lot of parsers will assume only up to a regular JS double. Just to be sure since I thought my change worked before, but somewhere along the line it was getting truncated to JS double range and losing a tiny bit of precision: `1234567890123456789` -> `1234567890123456800`

This seems to be the only place where parsing is done of the JSON, and the json-bigint uses bignumber.js instead of bigint like us.

Reply:
There is an option to get bigint instead of bignumber.js now that it's natively supported but I think we just read it all per-field into bigints throughout the connector code when we need to operate on it rather than converting the whole thing.

Reply:
@rooooooooob , I had to add parsing after all, yes, cuz I have been missing that built-in JSON stringification is also used to send messages thru the socket. Now these messages are being biging-stringified before sending and then bigint-parsed on the other side. Inside Yoroi itself `json-bigint` is configured to use native bigint wherever parsing is used, so we can make the typecheck for `bigint`.

@oskin1, when receiving messages back from Yoroi I have used a browserified version of the `json-bigint` without the native setting being set, so right now it will give out `bignumber.js` bigints, independent of what have been sent in (the PR is in a draft mode for now). There's a decision we need to make where we either will always return `bignumber.js` or always return native, or do some of it by default and then  we add some kind of a setting on the `ergo` object where the dev could do `ergo.setUsingNativeBigIntForParsing(true)`.